src/tree/GroupTreeItemBase.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "@microsoft/vscode-azext-utils";
+import { localize } from "../utils/localize";
+import { ResolvableTreeItem } from "./ResolvableTreeItem";
+import { ShallowResourceTreeItem } from "./ShallowResourceTreeItem";
+
+export abstract class GroupTreeItemBase extends AzExtParentTreeItem {
+    public readonly childTypeLabel: string = localize('resource', 'Resource');
+    public treeMap: { [key: string]: (ResolvableTreeItem | ShallowResourceTreeItem) } = {};
+    public abstract label;
+
+    public readonly cTime: number = Date.now();
+    public mTime: number = Date.now();
+
+    private _nextLink: string | undefined;
+
+    public hasMoreChildrenImpl(): boolean {
+        return !!this._nextLink;
+    }
+
+    public async loadMoreChildrenImpl(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        if (clearCache) {
+            // this.treeMap = {};
+        }
+
+        for (const ti of Object.values(this.treeMap)) {
+            if (ti instanceof ResolvableTreeItem) {
+                void ti.resolve(clearCache, context);
+            }
+        }
+
+        return <AzExtTreeItem[]><unknown>Object.values(this.treeMap);
```

Comment:
I'm not sure TS will let you, but if it does then IMO this is cleaner:
```suggestion
        return Object.values(this.treeMap) as AzExtTreeItem[];
```

Reply:
Yes, that works.